run.go

```diff
@@ -0,0 +1,358 @@
+	MessageCreation *StepDetailsMessageCreation `json:"message_creation,omitempty"`
+	ToolCalls       *StepDetailsToolCalls       `json:"tool_calls,omitempty"`
+}
+
+type StepDetailsMessageCreation struct {
+	MessageID string `json:"message_id"`
+}
+
+type StepDetailsToolCalls struct {
+	ToolCalls []ToolCall `json:"tool_calls"`
+}
+
+// RunStepList is a list of steps.
+type RunStepList struct {
+	RunSteps []RunStep `json:"data"`
+
+	httpHeader
+}
+
+// CreateRun creates a new run.
+func (c *Client) CreateRun(
+	ctx context.Context,
+	threadID string,
+	request RunRequest,
+) (response Run, err error) {
+	urlSuffix := fmt.Sprintf("/threads/%s/runs", threadID)
+	req, err := c.newRequest(ctx, http.MethodPost, c.fullURL(urlSuffix), withBody(request),
+		withBetaAssistantV1())
+	if err != nil {
+		return
+	}
+
+	err = c.sendRequest(req, &response)
+	return
+}
+
+// RetrieveRun retrieves a run.
+func (c *Client) RetrieveRun(
+	ctx context.Context,
+	threadID string,
+	runID string,
+) (response Run, err error) {
+	urlSuffix := fmt.Sprintf("/threads/%s/runs/%s", threadID, runID)
+	req, err := c.newRequest(ctx, http.MethodGet, c.fullURL(urlSuffix),
+		withBetaAssistantV1())
+	if err != nil {
+		return
+	}
+
+	err = c.sendRequest(req, &response)
+	return
+}
+
+// ModifyRun modifies a run.
+func (c *Client) ModifyRun(
+	ctx context.Context,
+	threadID string,
+	runID string,
+	request RunModifyRequest,
+) (response Run, err error) {
+	urlSuffix := fmt.Sprintf("/threads/%s/runs/%s", threadID, runID)
+	req, err := c.newRequest(ctx, http.MethodPost, c.fullURL(urlSuffix), withBody(request),
+		withBetaAssistantV1())
+	if err != nil {
+		return
+	}
+
+	err = c.sendRequest(req, &response)
+	return
+}
+
+// ListRuns lists runs.
+func (c *Client) ListRuns(
+	ctx context.Context,
+	threadID string,
+	limit *int,
+	order *string,
+	after *string,
+	before *string,
+) (response RunList, err error) {
+	urlValues := url.Values{}
+	if limit != nil {
+		urlValues.Add("limit", fmt.Sprintf("%d", *limit))
+	}
+	if order != nil {
+		urlValues.Add("order", *order)
+	}
+	if after != nil {
+		urlValues.Add("after", *after)
+	}
+	if before != nil {
+		urlValues.Add("before", *before)
+	}
+
+	encodedValues := ""
+	if len(urlValues) > 0 {
+		encodedValues = "?" + urlValues.Encode()
+	}
+
+	urlSuffix := fmt.Sprintf("/threads/%s/runs%s", threadID, encodedValues)
+	req, err := c.newRequest(ctx, http.MethodGet, c.fullURL(urlSuffix),
+		withBetaAssistantV1())
+	if err != nil {
+		return
+	}
+
+	err = c.sendRequest(req, &response)
+	return
+}
+
+// SubmitToolOutputs submits tool outputs.
+func (c *Client) SubmitToolOutputs(
+	ctx context.Context,
+	threadID string,
+	runID string,
+	request SubmitToolOutputsRequest) (response Run, err error) {
+	urlSuffix := fmt.Sprintf("/threads/%s/runs/%s/submit_tool_outputs", threadID, runID)
+	req, err := c.newRequest(ctx, http.MethodPost, c.fullURL(urlSuffix), withBody(request),
+		withBetaAssistantV1())
```

Comment:
nit: could we please either use single-line or all-in multiple lines, so here it would be 

```go
func (c *Client) SubmitToolOutputs(
	ctx context.Context,
	threadID string,
	runID string,
	request SubmitToolOutputsRequest,
) (response Run, err error) {
	urlSuffix := fmt.Sprintf("/threads/%s/runs/%s/submit_tool_outputs", threadID, runID)
	req, err := c.newRequest(
		ctx,
		http.MethodPost,
		c.fullURL(urlSuffix),
		withBody(request),
		withBetaAssistantV1(),
	)
```